src/main/java/com/example/cinematicketbookingapp/model/Movie.java

```diff
@@ -0,0 +1,25 @@
+package com.example.cinematicketbookingapp.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+@Entity
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder(toBuilder = true)
+@Data
```

Comment:
Przy encjach to trzeba uważać na takie cuda, zwłaszcza na equals/hashCode.

Reply:
Tak, wiem, że adnotacja `@Data` lub `@EqualsAndHashCode` nie są zalecane w przypadku encji, bo nieświadomie możemy stracić kontrolę np. podczas korzystania z kolekcji wykorzystujących hashe. Powinniśmy raczej sami decydować na podstawie jakich pól używamy `hashCode()` oraz `equals()` zamiast zdawać się na standardowe implementacje Lomboka. Najlepsze kwiatki to mogą wyjść podczas np. porównywania obiektów klas korzystających z kompozycji wykorzystując deep equals, wtedy możemy z łatwością wywołać `NullPointerException` jeśli jakieś pole jest `nullem` (przy standardowych dla Lomboka implementacjach `hashCode()` oraz `equals()`. Ponieważ encje mają swój unikalny identyfikator, to na jego podstawie powinien być wyliczany `hashCode()` oraz wywoływany `equals()` – oczywiście zakładając, że identyfikator nie zmieni nam się w trakcie wykonywania programu. Też często spotykam się z problemem standardowej implementacji `toString()` dla Lomboka, która zazwyczaj wywołuje `NullPointerException` podczas debugowania kodu. Przy okazji chciałbym zapytać odnośnie `toString()` – czy implementacja tej metody służy jedynie programiście i to ja powinienem ją zaimplementować w taki sposób żeby służyła mi np. podczas debugowania? Nie spotkałem się chyba jeszcze z sytuacją, w której ta metoda miałaby inne zastosowanie niż debugowanie lub wyświetlanie wyników w konsoli (dla prostych aplikacji).

Reply:
Tak, masz rację odnośnie toString. W zdecydowanej większości przypadków służy głównie do:
1. pisania obiektów do logów
2. diffów w testach gdy equals nie przechodzi